src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/ci/ciObjectFactory.java

```diff
@@ -75,11 +73,13 @@ public static ciMetadata getMetadata(Address addr) {
   }
 
   public GrowableArray<ciMetadata> objects() {
-    return GrowableArray.create(ciMetadataField.getValue(getAddress()), ciMetadataConstructor);
+    Address addr = getAddress().addOffsetTo(ciMetadataField.getOffset());
+    return GrowableArray.create(addr, ciMetadataConstructor);
   }
 
   public GrowableArray<ciSymbol> symbols() {
-    return GrowableArray.create(symbolsField.getValue(getAddress()), ciSymbolConstructor);
+    Address addr = getAddress().addOffsetTo(symbolsField.getOffset());
+    return GrowableArray.create(addr, ciSymbolConstructor);
```

Comment:
It's unclear to me why these two changes were needed. Don't they produce the same result before and after?

Reply:
When changing the fields from  `GrowableArray<..>*` to `GrowableArray<..>` I looked around and found another use of an embedded `GrowableArray` in `InlineTree.java` and took my cues from that. If you're certain these approaches are semantically identical I can drop these changes, but as I'm not exactly sure how to verify and test this I went the safe(?) route of leaning on prior art here.

Reply:
Ah, I see now. I didn't pick up on the change from a pointer type to an embedded type. Yes, I think your changes are correct and necessary. Rather than plucking the pointer value out of the field, you are now computing the address of the field, which seems like the right thing to be doing when changing the field from a pointer type to an embedded type.

Consider the SA changes reviewed. I'm not reviewing any of the hotspot changes.